examples/i2c_hal_ssd1306helloworld.rs

```diff
@@ -9,37 +9,29 @@ use panic_halt as _;
 
 use stm32f407g_disc as board;
 
-use ssd1306::{displayrotation::DisplayRotation, mode::TerminalMode, Builder, I2CDIBuilder};
+use core::fmt::Write;
 
-use crate::board::{hal::i2c::*, hal::prelude::*, hal::stm32};
+use ssd1306::{displayrotation::DisplayRotation, mode::TerminalMode, Builder, I2CDIBuilder};
 
-use core::fmt::Write;
+use crate::board::{hal::i2c::*, hal::pac, hal::prelude::*};
 
 #[cortex_m_rt::entry]
 fn main() -> ! {
-    if let Some(p) = stm32::Peripherals::take() {
+    if let Some(p) = pac::Peripherals::take() {
         let gpiob = p.GPIOB.split();
         let rcc = p.RCC.constrain();
 
         // Set up the clocks, going too fast exhibits some problem so let's take it slow for now
-        let clocks = rcc.cfgr.sysclk(40.mhz()).freeze();
+        let clocks = rcc.cfgr.sysclk(40.MHz()).freeze();
 
         // Set up the SCL pin of the I2C bus at PB6
-        let scl = gpiob
-            .pb6
-            .into_alternate_af4()
-            .internal_pull_up(true)
-            .set_open_drain();
+        let scl = gpiob.pb6.into_alternate_open_drain();
```

Comment:
Are there pull-up resistors on the board? If there are not, then:
```suggestion
        let scl = gpiob.pb6.into_alternate_open_drain().internal_pull_up(true);
```
or just
```
        let scl = gpiob.pb6.internal_pull_up(true);
```
If there are, then:
```
        let scl = gpiob.pb6;
```
is enough.

Reply:
Good question. Looks like they do:
<img width="222" alt="image" src="https://github.com/stm32-rs/stm32f407g-disc/assets/3221512/b6e62446-657d-44d1-bb95-ad8ef78390c1">
(screenshot from: https://www.st.com/resource/en/schematic_pack/mb997-f407vgt6-e01_schematic.pdf)

Thanks for the help: I have updated to:
```rust
        let scl = gpiob.pb6;
```